src/CMakeLists.txt

```diff
@@ -214,7 +220,7 @@ if (ENABLE_OPENMP)
 endif()
 
 if (ENABLE_HDF5)
-  target_link_libraries(parthenon PUBLIC HDF5_C)
+  target_link_libraries(parthenon INTERFACE $<TARGET_PROPERTY:HDF5_C,INTERFACE_LINK_LIBRARIES>)
```

Comment:
Why that change? Whenever you have to use `$<TARGET_PROPERTY:...>` you usually do something wrong.

Reply:
This might be my fault. @brryan I would revert this to what it was before unless it gives you problems. This might actually fix any HDF5 issues you have when building.

Reply:
@JoshuaSBrown Reverting this works fine for me, I haven't actually run into any HDF5 issues while building throughout this PR (I just have a `spack load hdf5` in my environment)